test/e2e/e2e_test.go

```diff
@@ -642,6 +642,68 @@ func checkMetricsCollection(t *testing.T, promClient prometheusv1.API, namespace
 	}
 }
 
+func runPromQuery(t *testing.T, promClient prometheusv1.API, query string, sampleTime time.Time) (float64, error) {
```

Comment:
nit: Maybe use a more descriptive name? This function has the specific purpose of returning a scalar result, so maybe something like `queryForScalar`?

Reply:
updated. please take a look and let me know if it needs some more refactoring